packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -63,11 +64,28 @@ export default class SaytDriverPlugin implements Plugin {
     fields: ['*'],
   }
 
-  constructor() {
+  /**
+   * Default product transformer identity function.
+   * Intended to be overwritten by passing a custom product transformer.
+   *
+   * @param product The product to be returned as-is.
+   * @returns The received product object.
+   */
+  transformProduct: ProductTransformer<P> = ((product: Record): Record => product) as any;
+
+  /**
+   * Binds relevant functions. Sets transformProduct property if a
+   * product transformer function is passed.
+   */
+  constructor(options?: SaytDriverOptions) {
     this.fetchAutocompleteTerms = this.fetchAutocompleteTerms.bind(this);
     this.fetchProductData = this.fetchProductData.bind(this);
     this.autocompleteCallback = this.autocompleteCallback.bind(this);
     this.searchCallback = this.searchCallback.bind(this);
+
+    if (options && options.productTransformer) {
```

Comment:
Instead of doing null checks everywhere, I would suggest defaulting `options` to an empty object in the parameter list, then assigning defaults through destructuring. Once the defaults are in place, we can assign them to their respective properties.

I realize we aren't doing this in the other plugins, but I think this is the cleaner pattern.

```typescript
constructor(options: SaytDriverOptions = {}) {
  const {
    productTransformer = ((product: Record): Record => product) as any,
  } = options;
  // ...

  this.transformProduct = productTransformer;
  // ...
}
```

If you want an additional layer of safety, you can check if `productTransformer` is a function before you assign it to the property.